public/js/app.js

```diff
@@ -0,0 +1,184 @@
+
+            let title = document.createElement("h3");
+            title.className = "title";
+            title.innerHTML = responseText[i].data.title;
+            article.appendChild(title);
+
+            let author = document.createElement("h6");
+            title.className = "author";
+            author.innerHTML = "by " + responseText[i].data.author + " &#xb7; " + moment.unix(responseText[i].data.created_utc).fromNow() + " &#xb7; " + responseText[i].data.ups + " views";
+            article.appendChild(author);
+
+            let selfText = document.createElement("p");
+            selfText.className = "selfText";
+            selfText.innerHTML = responseText[i].data.selftext.substring(0, 130) + "...";
+            article.appendChild(selfText);
+
+            let url = document.createElement("a");
+            url.className = "url";
+            url.href = responseText[i].data.url;
+            url.innerHTML = responseText[i].data.url.substring(0, 28);
+            url.target = '_blank';
+            article.appendChild(url);
+          }
+        });
+        nopixel.open("GET", "https://www.reddit.com/r/nopixel/.json");
+        nopixel.send();
+      }
+      if (selected === "RANDOM") {
+        articles.innerHTML = '';
+        const random = new XMLHttpRequest();
+        random.addEventListener("load", function() {
+          const responseText = JSON.parse(this.responseText).data.children;
+
+          for (let i = 0; i < responseText.length; i++) {
+            let article = document.createElement("div");
+            article.className = "article";
+            let articles = document.getElementById("content");
+            articles.appendChild(article);
+
+            let thumbnail = document.createElement('img');
+            thumbnail.className = "contentPictures";
+            thumbnail.src = responseText[i].data.thumbnail;
+            article.appendChild(thumbnail);
+
+            let title = document.createElement("h3");
+            title.className = "title";
+            title.innerHTML = responseText[i].data.title;
+            article.appendChild(title);
+
+            let author = document.createElement("h6");
+            title.className = "author";
+            author.innerHTML = "by " + responseText[i].data.author + " &#xb7; " + moment.unix(responseText[i].data.created_utc).fromNow() + " &#xb7; " + responseText[i].data.ups + " views";
+            article.appendChild(author);
+
+            let selfText = document.createElement("p");
+            selfText.className = "selfText";
+            selfText.innerHTML = responseText[i].data.selftext.substring(0, 130) + "...";
+            article.appendChild(selfText);
+          
+            let url = document.createElement("a");
+            url.className = "url";
+            url.href = responseText[i].data.url;
+            url.innerHTML = responseText[i].data.url.substring(0, 28);
+            url.target = '_blank';
+            article.appendChild(url);
```

Comment:
Instead of having 4 different event handlers that generally all do the same thing. You should be able to turn it into a function that can be used in multiple places instead.